src/org/nuxeo/openshift/library/ConnectClient.groovy

```diff
@@ -0,0 +1,87 @@
+package org.nuxeo.openshift.library
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.util.ArrayList
+import groovy.lang.Closure
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.client.CloseableHttpClient
+import org.apache.http.impl.client.HttpClientBuilder
+
+import static org.apache.http.auth.AuthScope.ANY
+
+def deployPackageToConnect(connectUsername, connectPassword, studioProject, packageFilePath, connectUrl='https://connect.nuxeo.com') {
+  def clientId = getClientIdByStudioProject(connectUsername, connectPassword, studioProject)
+  if (clientId == null) {
+    error("No clientId found with user '" + connectUsername + "' for Studio project '" + studioProject + "'.").
+    return
+  }
+
+  def uploadUrl = connectUrl + "/nuxeo/site/marketplace/upload?batch=true\\&project=${studioProject}\\&owner=${clientId}\\&client=${clientId}"
+  def format = "CONNECT_HTTP_STATUS: %{http_code}"
+  def packageFileArg = "package=@${packageFilePath}"
+  def connectResponseFile = "connect_response.txt"
+
+  sh "curl -w '${format}' -u ${connectUsername}:${connectPassword} -i -n -F ${packageFileArg} ${uploadUrl} > ${connectResponseFile}"
```

Comment:
why not using the httpclient here ?

Reply:
At the very beginning I started the implementation by trying to use the HttpClient. It was taking a bit of time, there were some issues. And then you told me to use a simple curl command, the one we are used to.